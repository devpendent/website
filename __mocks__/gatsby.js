const React = require('react')
const gatsby = jest.requireActual('gatsby')

module.exports = {
  ...gatsby,
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({ activeClassName, activeStyle, getProps, innerRef, ref, replace, to, ...rest }) =>
      React.createElement('a', {
        ...rest,
        href: to
      })
  ),
  StaticQuery: jest.fn(),
  graphql: jest.fn(),
  useStaticQuery: jest.fn()
}
