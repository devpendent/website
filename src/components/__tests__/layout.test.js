import { useStaticQuery } from 'gatsby'
import React from 'react'
import { render } from 'react-testing-library'
import Layout from '../layout'

describe('Layout', () => {
  beforeEach(() => {
    useStaticQuery.mockReturnValueOnce({
      site: {
        siteMetadata: {
          desciption:
            'An Open Sourced Platform for Indonesia Election Real Count',
          title: 'Devpendent'
        }
      }
    })
  })

  it('renders correctly', () => {
    const { container } = render(
      <Layout>
        <h1>Devpendent Website</h1>
      </Layout>
    )
    expect(container).toMatchSnapshot()
  })
})
