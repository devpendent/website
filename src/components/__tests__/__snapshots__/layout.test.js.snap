// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<div>
  <header
    style="background: rebeccapurple; margin-bottom: 1.45rem;"
  >
    <div
      style="margin: 0px auto; max-width: 960px; padding: 1.45rem 1.0875rem;"
    >
      <h1
        style="margin: 0px;"
      >
        <a
          href="/"
          style="color: white; text-decoration: none;"
        >
          Devpendent
        </a>
      </h1>
    </div>
  </header>
  <div
    style="margin: 0px auto; max-width: 960px; padding: 0px 1.0875rem 1.45rem 1.0875rem;"
  >
    <main>
      <h1>
        Devpendent Website
      </h1>
    </main>
    <footer>
      Â© 
      2019
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>
</div>
`;
